src/main/java/org/myrobotlab/framework/Service.java

```diff
@@ -2813,5 +2815,15 @@ public boolean isPeerStarted(String peerKey) {
   protected void registerForInterfaceChange(Class<?> clazz) {
     Runtime.getInstance().registerForInterfaceChange(getClass().getCanonicalName(), clazz);
   }
+  
 
+  /**
+   * default emtpy getDefaultConfig - will be overriden by services which provide sane defaults
+   * @return
+   */
+ static public Map<String, ServiceConfig> getDefaultConfig() {
+   Map<String, ServiceConfig> ret = new HashMap<>();
```

Comment:
i feel like returning null is better than returning an empty config..

Reply:
a bunch of npe safety checks then ... not sure i feel the same way